plugins/woocommerce/includes/tracks/class-wc-site-tracking.php

```diff
@@ -93,6 +93,16 @@ public static function add_tracking_function() {
 					delete( eventProperties._ui );
 					delete( eventProperties._ut );
 				}
+				// Verify the event name is correct
+				if ( ! /^(([a-z0-9]+)_){2}([a-z0-9_]+)$/.test( eventName ) ) {
+					throw new Error( `A valid event name must be specified. The event name: "${ eventName }" is not valid.` );
```

Comment:
Sorry to drive-by this PR but I noticed that you're throwing an error here, which is a **breaking change** for calling this method and may lead to "white page" kind of errors for users if callers of the method originally didn't expect this would ever throw? I looked at existing code and it doesn't have catches around it, so are you sure you'd want that to happen in prod? I think that could lead any mistakes in tracking to break the app for users.

I think you'd have to update every call to `recordEvent` to be wrapped in a try and maybe only throw in development or perhaps change this code to only throw in development and console.error in dev perhaps?

Reply:
Thank you @samueljseay for your feedback. I think that using a `console.log` would be enough in this case. I added that change in the commit d099918